peeringdb_server/views.py

```diff
@@ -2820,7 +2823,10 @@ def done(self, form_list, **kwargs):
 
         user_language = self.get_user().get_locale()
         translation.activate(user_language)
+        success_url = self.get_success_url()
         response = redirect(self.get_success_url())
+        if success_url == "/oauth2/authorize/":
+            response.set_signed_cookie("oauth_session", self.request.user, max_age=1800)
```

Comment:
max_age should come from a settings variable, and the settings variable (via `set_option`)